src/BuildScriptGeneratorCli/Commands/BuildScriptCommand.cs

```diff
@@ -4,31 +4,110 @@
+            this.PlatformVersion = input.PlatformVersion;
+            this.ShouldPackage = input.ShouldPackage;
+            this.OsRequirements = input.OsRequirements;
+            this.AppType = input.AppType;
+            this.BuildCommandsFileName = input.BuildCommandsFileName;
+            this.CompressDestinationDir = input.CompressDestinationDir;
+            this.Properties = input.Properties;
+            this.DynamicInstallRootDir = input.DynamicInstallRootDir;
+            this.LogFilePath = input.LogFilePath;
+            this.DebugMode = input.DebugMode;
+        }
 
-        [Option(
-            "--output",
-            CommandOptionType.SingleValue,
-            Description = "The path that the build script will be written to. " +
-                          "If not specified, the result will be written to STDOUT.")]
         public string OutputPath { get; set; }
 
+        public static Command Export(IConsole console)
+        {
+            var logOption = new Option<string>(OptionTemplates.Log, OptionTemplates.LogDescription);
+            var debugOption = new Option<bool>(OptionTemplates.Debug, OptionTemplates.DebugDescription);
+            var sourceDirArgument = new Argument<string>(
+                name: "SourceDir",
+                description: "The source directory.",
+                getDefaultValue: () => Directory.GetCurrentDirectory());
+            var platformOption = new Option<string>(OptionTemplates.Platform, OptionTemplates.PlatformDescription);
+            var platformVersionOption = new Option<string>(OptionTemplates.PlatformVersion, OptionTemplates.PlatformVersionDescription);
+            var packageOption = new Option<bool>(OptionTemplates.Package, OptionTemplates.PackageDescription);
+            var osReqOption = new Option<string>(OptionTemplates.OsRequirements, OptionTemplates.OsRequirementsDescription);
+            var appTypeOption = new Option<string>(OptionTemplates.AppType, OptionTemplates.AppTypeDescription);
+            var buildCommandFileNameOption = new Option<string>(OptionTemplates.BuildCommandsFileName, OptionTemplates.BuildCommandsFileNameDescription);
+            var compressDestDirOption = new Option<bool>(OptionTemplates.CompressDestinationDir, OptionTemplates.CompressDestinationDirDescription);
+            var propertyOption = new Option<string[]>(OptionTemplates.Property, OptionTemplates.PropertyDescription);
+            var dynamicInstallRootDirOption = new Option<string>(OptionTemplates.DynamicInstallRootDir, OptionTemplates.DynamicInstallRootDirDescription);
+            var buildScriptOutputOption = new Option<string>(
+                name: "--output",
+                description: "The path that the build script will be written to. " +
+                          "If not specified, the result will be written to STDOUT.");
+
+            var command = new Command("build-script", "Generate build script to standard output.")
```

Comment:
Nit: Is there a reason we changed this from a constant to hard-coded? I'd prefer to put these back into constants in this file or a general constants file that has all command names